src/lib/storage/base_dictionary_segment.hpp

```diff
@@ -50,5 +50,7 @@ class BaseDictionarySegment : public AbstractEncodedSegment {
    * @brief Returns encoding specific null value ID
    */
   virtual ValueID null_value_id() const = 0;
+
+  virtual bool is_nullable() const = 0;
```

Comment:
Rename to `has_nulls` (or contains null values as used below)? Given the segment's immutability, we should remove that element of doubt.

Reply:
I know, it's on my list but I did not want to proceed until we have figured out if we really would want this.

Reply:
What about `does_not_contain` similar to our filters. I think that better conveys the fact that we do usually not know if there are any NULLs or not. But if this is true, we guarantee that there are no NULLs.

Reply:
works for me